### PR TITLE
lib: bsdlib: add option for manual initialization

### PR DESCRIPTION
**Pull request by lemrey**
Added an option to let bsdlib be initialized manually.
Added `net/bsdlib.h` interface, for use by the application.

- `bsdlib.h` -> used by the application
- `bsd_os.h` -> used by the library to glue onto the OS

Sample's behavior left unchanged, since by default he library is initialized during `SYS_INIT`.

### CODE REVIEW

File: include/net/bsdlib.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+/**
+ * @brief Initialize bsdlib.
+ *
+ * This function is synchrounous and it will block for
+ * --very long-- when the modem firmware is being updated.
```
Comment (rlubos) [inline include/net/bsdlib.h]: `very long` does not provide too much information, perhaps we could provide an approximate value here? A few minutes perhaps?

File: lib/bsdlib/bsdlib.c [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+#include <bsd.h>
+#include <bsd_platform.h>
+
+extern void ipc_proxy_irq_handler(void);
+
+static int _bsdlib_init(struct device *unused)
+{
+	/* Setup the network IRQ used by the BSD library.
+	 * Note: No enable irq_enable here, that is done through bsd_init.
```
Comment (rlubos) [inline lib/bsdlib/bsdlib.c]: `No enable irq_enable here` sounds odd, perhaps you meant `No call to irq_enable here`?

File: lib/bsdlib/CMakeLists.txt [hunk 3]
```diff
@@ -4,7 +4,7 @@
 # SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
 #
 zephyr_library()
+zephyr_library_sources(bsdlib.c)
```
Comment (rlubos) [inline lib/bsdlib/CMakeLists.txt]: I wonder about `bsd_sanity.c` - now as we have a dedicated `bsdlib.c` source file, could we move `bsd_sanity.c` contents into it? All in all, it's just a check, I don't think it requires a separate source file.

File: include/net/bsdlib.h (hunk 1 above)
Comment (lemrey) [reply include/net/bsdlib.h]: Updated and extended.

File: lib/bsdlib/bsdlib.c (hunk 2 above)
Comment (lemrey) [reply lib/bsdlib/bsdlib.c]: Updated old comment.

File: lib/bsdlib/CMakeLists.txt (hunk 3 above)
Comment (lemrey) [reply lib/bsdlib/CMakeLists.txt]: Moved inside `bsdlib.c`

**Pull request merged**